src/open_samus_returns_rando/lua_editor.py

```diff
@@ -215,13 +215,15 @@ def _create_custom_init(self, editor: PatcherEditor, configuration: dict) -> str
             max_life += etanks * energy_per_tank
 
         # use _MAX if main is unlocked to unlock the ammo too
-        # FIXME: These checks are kinda wrong if you use something like `"ITEM_WEAPON_MISSILE_LAUNCHER": 0`
-        if "ITEM_WEAPON_MISSILE_LAUNCHER" in inventory and "ITEM_MISSILE_TANKS" in inventory:
-            inventory["ITEM_WEAPON_MISSILE_MAX"] = inventory.pop("ITEM_MISSILE_TANKS")
-        if "ITEM_WEAPON_SUPER_MISSILE" in inventory and "ITEM_SUPER_MISSILE_TANKS" in inventory:
-            inventory["ITEM_WEAPON_SUPER_MISSILE_MAX"] = inventory.pop("ITEM_SUPER_MISSILE_TANKS")
-        if "ITEM_WEAPON_POWER_BOMB" in inventory and "ITEM_POWER_BOMB_TANKS" in inventory:
-            inventory["ITEM_WEAPON_POWER_BOMB_MAX"] = inventory.pop("ITEM_POWER_BOMB_TANKS")
+        LAUNCHER_TANK_MAPPING = [
+            ("ITEM_WEAPON_MISSILE_LAUNCHER", "ITEM_MISSILE_TANKS"),
+            ("ITEM_WEAPON_SUPER_MISSILE", "ITEM_SUPER_MISSILE_TANKS"),
+            ("ITEM_WEAPON_POWER_BOMB", "ITEM_POWER_BOMB_TANKS"),
+        ]
+        for launcher, tanks in LAUNCHER_TANK_MAPPING:
+            if inventory.get(launcher) and tanks in inventory:
```

Comment:
Either your PR title is wrong or your code. 
PR title says `... if < 0` but this check only prevents it if it is equal to zero or the key is not present.
Don't know why anyone would ever write such a thing but `"ITEM_WEAPON_MISSILE_LAUNCHER": -1` would still do the inventory update.

Reply:
😐